### ISSUE TITLE
Include and Exclude end devices by sending MQTT payloads on topics

### ISSUE THREAD

**Issue by bilaldpl123** [zwave-js/zwavejs2mqtt#2652]
is it possible to include/exclude end devices by through MQTT ? If yes where can I find information related to required topics header and payload?

**Comment by trankin** [comment]
I think what you are looking for is in the MQTT apis: https://zwave-js.github.io/zwavejs2mqtt/#/guide/mqtt?id=apis

**Comment by bilaldpl123** [comment]
Hi,

Thanks for the quick reply. I have already checed it but I have not able to find information about what should be a mqtt message payload for these different mqtt API? 

Also, I tried this topic to get node info without any payload but I didn't get any response? 

<mqtt_prefix>/<?node_location>/<node_name>/nodeinfo

what should be a payload value for this?

**Comment by trankin** [comment]
Here is an example API call I made to start inclusion:

zwave/_CLIENTS/ZWAVE_GATEWAY-Zwavejs2Mqtt/api/startInclusion/set

{
  "args": [
    0,
    {
      "forceSecurity": false
    }
  ]
}

Response:
{"success":true,"message":"Success zwave api call","result":true,"args":[0,{"forceSecurity":false}],"origin":{"args":[0,{"forceSecurity":false}]}}

**Comment by robertsLando** [comment]
You can also use `provisionSmartStartNode` with the result of a qr code scan in order to do an SS inclusion.

**Comment by bilaldpl123** [comment]
This is great. I will try that. Is it still also possible to add DSK or QR code to SmartStart list through MQTT? 

We are thinking to remove UI part and use only MQTT interface to communicate with z-wave JS and manage end devices. We will write MQTT wrapper which will manage this and communicate with backend cloud. What do you think. Is this possible?

**Comment by robertsLando** [comment]
> We are thinking to remove UI part and use only MQTT interface to communicate with z-wave JS and manage end devices. We will write MQTT wrapper which will manage this and communicate with backend cloud. What do you think. Is this possible?

It is possbile yeah, the only 'hard' thing is handling the devices management. I would use the provided UI for that as it's really hard to handle

**Comment by bilaldpl123** [comment]
Yeah we believe that we can do device management with MQTT wrapper that we will write with assumption that the MQTT broker will retain all end devices related information and whenever any MQTT client will connect with local MQTT broker, it will get all those messages and by using those message we can manage end devices. 

My only concern is if we remove UI part and use only MQTT to communicate with end devices through Zwave JS, will it work? 

Does Zwave JS stores information related to included devices or GUI application is responsible for that?

**Comment by robertsLando** [comment]
> My only concern is if we remove UI part and use only MQTT to communicate with end devices through Zwave JS, will it work?

Question is how do you edit UI settings without UI? Like logging, controller path etc... 

> I just noticed that we don't get any message on MQTT when a device is removed from the network. I subscribed to wildcard topic zwave/# but didn't get any MQTT message after device is removed.

You should enable sending mqtt events to mqtt but as I said that is not the best choice.

As an alternative you could use the integrated zwave-js-server by enabling it in settings and use it's websocket to handle everything like Home-Assistant team is doing right now

**Comment by bilaldpl123** [comment]
My understanding was that when user peform any action on UI for settings there must be any JS API which get called. So I was thinking if we may trigger those APIs through MQTT messages by changing mqtt client part of Zwave JS. 

Let me know if I am wrong. Sorry I am not an expert in JS, backend or front end. I am embedded software developer .
